src/Psy/TabCompletion/Matchers/AbstractMatcher.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace Psy\TabCompletion\Matchers;
+
+use Psy\Context;
+use Psy\ContextAware;
+
+/**
+ * Class AbstractMatcher
+ * @package Psy\TabCompletion\Matchers
+ */
+abstract class AbstractMatcher implements ContextAware
+{
+    /** Syntax types */
+    const CONSTANT_SYNTAX = '^[A-Z0-9-_]+$';
+    const VAR_SYNTAX = '^\$[a-zA-Z0-9-_]+$';
```

Comment:
Nope, this is a valid name...

> Variable names follow the same rules as other labels in PHP. A valid variable name starts with a letter or underscore, followed by any number of letters, numbers, or underscores. As a regular expression, it would be expressed thus: '[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*'

I'm refering to the text before the note